### ISSUE TITLE
NixOS support for multiple X sessions

### ISSUE THREAD

**Issue by deliciouslytyped** [NixOS/nixpkgs#83646]
I'm writing this in a fit of rage so maybe I can come back and fix the issue up a bit later;
Given how great we are, why can't we support a sane way of starting multiple X sessions?
Ever since the switch to systemd I've not seen any of this work anywhere, though other linuxes seem to support it somehow.

Currently I just tried `nix-shell -p xorg.xinit --run startx`, which resulted in all input to my machine besides the power button hanging and is reproducible in the 19.09 graphical iso.
If I press the power button the machine goes into a normal shutdown path.
I cannot switch TTYs in the hung state, but I can normally. The machine appears to continue computing, just all input hangs. (How is that even possible outside X11?)

**Comment by deliciouslytyped** [comment]
This could also be entirely my fault and lack of knowledge on how the stupid X11 startup system works, but back in the day I could just run `startx` in any TTY and it would work just fine.

**Comment by deliciouslytyped** [comment]
Not surprisingly, some things exist; `services.xserver.exportConfiguration = true;` somehow prevents startx from self mutilation by cutting all its ownlimbs off and gouging it's own eyes out - which would be a nice behaviour to avoid in any case.

Or rather, the "somehow" raises the question of why startx with no configuration is so horrible that it gives the system the opposite of locked-in syndrome.

**Comment by deliciouslytyped** [comment]
![Selection_105](https://user-images.githubusercontent.com/47436522/77843874-9a435a00-71a1-11ea-87a2-00f24e4e22e9.png)
This screen will mock you in all it's glory if you have the sheer audacity of running `startx` and will laugh in your face as you hope it will accept the external USB devices you are trying to plug in - after going through several 4D rotations you will beg and plead but there will be no answer. The fans will hum - it's not a hang. All you are left with is defeat and a reboot - after all, you hold the power - but only the power.

**Comment by deliciouslytyped** [comment]
Even with `services.xserver.exportConfiguration = true;` it looks like it still kills the previous X session though.

**Comment by deliciouslytyped** [comment]
Even with `services.xserver.exportConfiguration = true;` it looks like it still kills the previous X session though.

**Comment by primeos** [comment]
Just a quick FYI for debugging/testing: You can recover your control over the keyboard by pressing SysRq+R (see [Magic SysRq key](https://en.wikipedia.org/wiki/Magic_SysRq_key) - that helped me a lot while setting up `startx`).

Currently I use the following basic configuration (but I only tested one session so far and don't use it much - only as a real X11 fallback (vs. XWayland)):
```
  services.xserver = {
    enable = true;
    autorun = false;
    displayManager.startx.enable = true;
    exportConfiguration = true;
    libinput.enable = true;
  };
```

Regarding the killing of previous sessions: Maybe something like `startx -- :1` works (it might default to the display `:0` and override the socket `/tmp/.X11-unix/X0`).

IMO we should at least add some more documentation to `services.xserver.displayManager.startx.enable` (e.g. mention SysRq and the issues without a correct configuration, link to the wiki page, etc.).

**Comment by deliciouslytyped** [comment]
Yes I tried that and the entire REISUB mnemonic didn't do a damn thing.

**Comment by deliciouslytyped** [comment]
So...some pretty big user error involved here...
I didn't know SysRq-s could be turned off originally, and I'd only been confirming with alt-sysrq-h that they are enabled. It turns out the -b sysrq is disabled by default, for example....

**Comment by deliciouslytyped** [comment]
xlink; https://github.com/NixOS/nixpkgs/issues/83694#issuecomment-605657381

**Comment by deliciouslytyped** [comment]
It looks like I forgot to follow up on this, I mostly solved my problem by discovering that SDDM can handle multiple sessions by itself, if you do the "switch user" dance. This may have some limitations but it's a start.

**Comment by deliciouslytyped** [comment]
Probably warrants some followup after the new services RFC.

**Comment by wedens** [comment]
Do you use multiple X servers with the same Xorg configs? I'm interested in a way to have multiple servers with different configs (so that servers use different GPUs).

**Comment by flokli** [comment]
I'm not really sure what's still left for this issue. Everything in the initial description is strike-through.

@deliciouslytyped, can you update the description, so it's easy to grasp what's the issue here?
